packages/frontend/src/contexts/User.js

```diff
@@ -66,6 +66,8 @@ class User {
 
   async loadReputation() {
     const epoch = this.userState.calcCurrentEpoch()
+    const blockNumber = await provider.getBlockNumber()
+    await this.userState.waitForSync(blockNumber)
```

Comment:
Why do we need to `waitForSync` here? It shouldn't affect the epoch or rep loading?

Reply:
I use `loadReputation` in `requestReputation` and `stateTransition` (both affect `getRepByAttester(epoch)`)
so if `userState` hasn't been synced, the `getRepByAttester` is not updated
but yeah if it is in `signup` it doesn't affect rep (in current version)

Reply:
Right, we need `waitForSync` in those functions specifically because they're waiting on a new transaction, but we don't need it to load the reputation generally. It's not something that is breaking, but if for example, we had a loop loading the reputation to update the UI we wouldn't need `waitForSync`.

Also line 69 is redundant, when `waitForSync` is called with no arguments it loads the latest block and uses that.